api/src/test/java/jakarta/data/metamodel/AttributeTest.java

```diff
@@ -110,7 +110,7 @@ void shouldCreateNotInRestriction() {
 
     @Test
     void shouldThrowExceptionForEmptyNotInRestriction() {
-        assertThatThrownBy(() -> testAttribute.notIn(Set.of()))
+        assertThatThrownBy(() -> testAttribute.notIn((Set<String>) null))
```

Comment:
A test that says it covers empty should actually pass in empty, not null,
```suggestion
        assertThatThrownBy(() -> testAttribute.notIn(Set.of()))
```